source/patterns/@aws-solutions-constructs/aws-lambda-sagemakerendpoint/README.md

```diff
@@ -0,0 +1,132 @@
+# aws-lambda-sagemakerendpoint module
+
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**: | <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span> |
+| :--------------------------- | :------------------------------------------------------------------------------------------------ |
+
+
+<div style="height:8px"></div>
+
+| **Language**                                                                                   | **Package**                                                      |
+| :--------------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| ![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python             | `aws_solutions_constructs.aws_lambda_sagemakerendpoint`          |
+| ![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript | `@aws-solutions-constructs/aws-lambda-sagemakerendpoint`         |
+| ![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java                   | `software.amazon.awsconstructs.services.lambdasagemakerendpoint` |
+
+This AWS Solutions Construct implements an AWS Lambda function connected to an Amazon SageMaker Endpoint.
+
+Here is a minimal deployable pattern definition in Typescript:
+
+```typescript
+import { LambdaToSqs, LambdaToSqsProps } from '@aws-solutions-constructs/aws-lambda-sagemakerendpoint';
+
+// Create IAM Role to be assumed by SageMaker. You may want to scope down the Role permissions
+const sagemakerRole = new iam.Role(stack, 'SagemakerRole', {
```

Comment:
The minimum deployable code should not require the client to create a role - this is the kind of default stuff that Constructs takes care of. 

It appears this is required if an existing endpoint is not provided, this should not be the case. AFAIK no other construct takes a role as an property, yet all use roles.

Reply:
My two reasons for this are:

1. When the client wants to create a new Sagemaker Endpoint, he has to pass `executionRoleArn` with the `modelProps?` [sagemaker.CfnModelProps](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sagemaker.CfnModelProps.html). It is not optional.
2. Since, the role is not created inside the construct (because of reason 1), a dependency must be created between the Sagemaker model and Role. 
```
    // Create the Sagemaker's Model
    model = new sagemaker.CfnModel(scope, 'SagemakerModel', finalModelProps);
    // Add dependency on the Sagemaker's role
    model.node.addDependency(role);
 ```